src/realm/sync/instruction_replication.cpp

```diff
@@ -487,7 +495,9 @@ void SyncReplication::add_int(const Table* table, ColKey col, ObjKey ndx, int_fa
 
 void SyncReplication::set(const Table* table, ColKey col, ObjKey key, Mixed value, _impl::Instruction variant)
 {
-    Replication::set(table, col, key, value, variant);
+    if (!table->is_asymmetric()) {
```

Comment:
What is the motivation for not calling `Replication::set` here? An if it is because we want to avoid notifications then why not have the same test when updating collections?

Reply:
wouldn't it be an error since the object is never created in the realm and hence there is nothing to update?

Reply:
You have a point. I may need to do it for collections too. I will look into it.

Context: The reason I decided to handle object "deletion" at the replication level is because some SDKs don't use `Object::create` but `Table::create` directly. I initially deleted the object before Object::create returned (with the idea that the server would ignore a delete instruction on an asymmetric object), but this is not enough for all SDKs.

Reply:
It turns out it's not needed. I'll remove these checks.